### PR TITLE
Unify instance initializers.

### PR DESCRIPTION
**Pull request by cibernox**
In fastboot 1.0 the concept of environment-exclusive initializers is gone. Instead,
addons are encouraged to use the existence of the `Fastboot` constant to know about
the environment the app is running on.

This PR unifies the fastboot/head.js initializers into the main one

### CODE REVIEW

File: app/instance-initializers/head.js [hunk 1]
```diff
@@ -1,30 +1,33 @@
...
     }
-    document.head.removeChild(startMeta);
-    document.head.removeChild(endMeta);
-  }
 
-  const component = owner.lookup('component:head-layout');
-  component.appendTo(document.head);
+    const component = owner.lookup('component:head-layout');
+    component.appendTo(document.head);
+  } else {
```
Comment (kratiahuja) [inline app/instance-initializers/head.js]: You are merging the fastboot initializer here and sending the extra bytes to browser that will never be run. Ideally you can move the new existing `app/instance-initializer/fastboot/head.js` to `fastboot/instance-initializers/head.js`. That's how we envisioned apps moving. Also currently the rc builds auto migrate the fastboot initializers to their new location with throwing a warning asking addons to the do the same. 

I would suggest you to take that approach and then this addon does a major version bump.

Comment (cibernox) [inline app/instance-initializers/head.js]: I misunderstood the instructions then. I'll update it.

Comment (bcardarella) [inline app/instance-initializers/head.js]: I'm really confused on what the suggested way to handle fastboot/browser code is. Originally it was separate builds, then it was a single build with flags guarding, now it seems there will be separate builds again?

Comment (cibernox) [inline app/instance-initializers/head.js]: From slack conversation, initializers in `/app/(instance-)initializers` will run both in fastboot and the browser, so you have to guard if they do fastboot-incompatible stuff.

Initializers in `/fastboot/(instance-)initializers` will only run in fastboot so there is no need to guard.

Comment (scottmessinger) [inline app/instance-initializers/head.js]: @bcardarella There's one build that runs in both browser & node. There can be an additional file(s) that are loaded only in fastboot. The files that are loaded only in fastboot are loaded by adding the file to the fastboot manifest. For more, check out: https://github.com/ember-fastboot/ember-cli-fastboot/issues/413

Comment (kratiahuja) [inline app/instance-initializers/head.js]: > I'm really confused on what the suggested way to handle fastboot/browser code is. Originally it was separate builds, then it was a single build with flags guarding, now it seems there will be separate builds again?

There are still no separate builds now. We are building the additional asset containing such fastboot overrides. For the fastboot initializers we never asked to guard against the flags. It's only for the browser initializers that you need to move from `app/initializers/browser/somefile.js` to `app/initializers/somefile.js` and put the guard. For the `app/initializers/fastboot/somefile.js` to `fastboot/initializers/somefile.js`.

We are working on updating the fastboot website to talk about the new scheme and new structure. There is also a cheatsheet that talks about the various migration cases.

**Comment by cibernox** [comment]
I close this, superseded by #32

File: app/instance-initializers/head.js (hunk 1 above)
Comment (scottmessinger) [inline app/instance-initializers/head.js]: @kratiahuja I might be misunderstanding things, but by adding a file to the fastboot manifest, aren't you creating a file that is loaded only in fastboot?

Comment (bcardarella) [inline app/instance-initializers/head.js]: > aren't you creating a file that is loaded only in fastboot

this is my understanding as well, which would definitely seem like a separate build. Outside the scope of discussion for this repo but the churn on Fastboot 1.0 is very difficult to keep up on.

Comment (kratiahuja) [inline app/instance-initializers/head.js]: > I might be misunderstanding things, but by adding a file to the fastboot manifest, aren't you creating a file that is loaded only in fastboot?

@scottmessinger yes. The usecase that you are talking about earlier is if you have to import external libraries in fastboot environment. The usecase in this PR is different than that are refers to case 4 in the cheatsheet. 

> Outside the scope of discussion for this repo but the churn on Fastboot 1.0 is very difficult to keep up on.

@bcardarella Have you seen the migration cheatsheet? Would love your feedback on the issue Scott posted with any additional usecases that seem confusing for you and we can work with making things better. Any help to make it better is greatly appreciated. 

Things will be much simpler to understand once the website is updated.

**Pull request closed without merging**